### DISCUSSION TITLE
线程数飙升问题

### DISCUSSION THREAD

**Discussion by githubzhangfei** [apache/rocketmq#7435] [Q&A]
项目是基于rocketmq做的一个数据处理引擎，现在是单机运行，客服端有50个topic，当数据来的时候，当前节点处理完后会丢到下个节点，每个节点都是一个topic。现在问题是我启动这50个监听的时候，线程数会达到1800左右，不启动的时候是80个左右。想知道这正常吗？中午我会贴出来核心代码

**Comment by cserwen** [comment]
每个 topic 是否使用了一个独立的pushConsumer去消费？ 

单个PushConsumer 启动时会设置20个消费线程，你可以修改这个参数来控制线程数

**Reply by githubzhangfei** [reply]
我是设置的最小消费线程是2个最大20个。新增一个topic 根据观察发现会新增24个线程。不知道这种是否正常

**Comment by figuewang** [comment]
线程数和queue的数量也成倍数关系，我理解是消费者数量少了，多加几个，或者把topic的queue的数量调小
